packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -386,6 +393,7 @@ export class SharedTreeBranch<
 		this.emit("beforeChange", changeEvent);
 		this.head = startCommit;
 		this.emit("afterChange", changeEvent);
+		this.emit("transactionRolledBack", this.transactions.size === 0);
```

Comment:
I think this event should also be emited in the `if (commits.length === 0) {` block above.

Reply:
You would have caught this if you had replicated the test coverage of that the other events have. Please add that coverage.